src/nacl/bindings/crypto_pwhash_scryptsalsa208sha256.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2013 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function
+
+from nacl._sodium import ffi, lib
+from nacl.exceptions import CryptoError
+
+
+crypto_pwhash_scryptsalsa208sha256_SALTBYTES = \
+    lib.crypto_pwhash_scryptsalsa208sha256_saltbytes()
+crypto_pwhash_scryptsalsa208sha256_STRBYTES = \
+    lib.crypto_pwhash_scryptsalsa208sha256_strbytes()
+crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_INTERACTIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_opslimit_interactive()
+crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_INTERACTIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_memlimit_interactive()
+crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_SENSITIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_opslimit_sensitive()
+crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_SENSITIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_memlimit_sensitive()
+
+OPSLIMIT_INTERACTIVE = \
+    crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_INTERACTIVE
+MEMLIMIT_INTERACTIVE = \
+    crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_INTERACTIVE
+OPSLIMIT_SENSITIVE = \
+    crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_SENSITIVE
+MEMLIMIT_SENSITIVE = \
+    crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_SENSITIVE
+SALTBYTES = \
+    crypto_pwhash_scryptsalsa208sha256_SALTBYTES
+STRBYTES = \
+    crypto_pwhash_scryptsalsa208sha256_STRBYTES
+
+
+def crypto_pwhash_scryptsalsa208sha256(outlen, passwd, salt,
+                                       opslimit=OPSLIMIT_SENSITIVE,
+                                       memlimit=MEMLIMIT_SENSITIVE
+                                       ):
+    """
+    returns uses the ``passwd`` and ``salt`` to produce derive a key
+    of ``outlen`` bytes can be tuned by picking different
+    ``opslimit`` and ``memlimit``.
+
+    The constants
+        - :py:const:`.OPSLIMIT_INTERACTIVE`
+        - :py:const:`.MEMLIMIT_INTERACTIVE`
+        - :py:const:`.OPSLIMIT_SENSITIVE`
+        - :py:const:`.MEMLIMIT_SENSITIVE`
+
+    are provided as a guidance for correct settings respectively for the
+    interactive login and the long term key protecting sensitive data
+    usage cases.
+
+    :param outlen: int
+    :param passwd: bytes
+    :param salt: bytes  *must* be *exactly* :py:const:`.SALTBYTES` long
+    :param opslimit: int
+    :param memlimit: int
+    :rtype: bytes
+    """
+
+    if len(salt) != SALTBYTES:
```

Comment:
I think this would be an excellent fit for the utils.ensure() assert replacer from #223